### DISCUSSION TITLE
[Kavita] Swipe pagination on epub reader

### DISCUSSION THREAD

**Discussion by YodaDaCoda** [Kareadita/Kavita#2666] [Ideas]
### Idea Description

Tap pagination is all well and good, but sometimes I wanna swipe.

Specifically, I have a little bluetooth page turner that works by acting as an input device that emulates swipes on the screen. It doesn't work with Kavita, but does work with other ereader apps that allow swipe-to-paginate.

### Duration of Using Kavita

8 months

### Idea Category

User Experience

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by lyingloverr** [comment]
There is swipe pagination?

![image](https://github.com/Kareadita/Kavita/assets/108381076/ab951dd4-063d-4026-af39-6977366c5b0c)

**Comment by scare376** [comment]
I'm guessing you mean for the epub reader as there is already swipe pagination in the archive/image reader. Please update your discussion to specify which reader you are hoping to include this on.

**Reply by YodaDaCoda** [reply]
Yep didn't realise the different readers were different components, I really only use kavita for epubs

**Reply by lyingloverr** [reply]
ahahah sorry for my misunderstanding xD

**Comment by ShivamAmin** [comment]
> Specifically, I have a little bluetooth page turner that works by acting as an input device that emulates swipes on the screen. It doesn't work with Kavita, but does work with other ereader apps that allow swipe-to-paginate.

I've been looking into hardware page turning, would love to know what you're using and what platform

**Reply by YodaDaCoda** [reply]
I got [this](https://www.amazon.com.au/gp/product/B09X5JM93P) jibber, but there's heaps like it. Works nicely with Android, though you gotta double-tap the button to get left/right swipe (single-tap action is up/down swipe).

**Comment by majora2007** [comment]
There is a draft PR but needs some serious work, if anyone wants to take the mantle up and drive this home:
https://github.com/Kareadita/Kavita/pull/2754
